graph/topo/johnson_cycles_test.go

```diff
@@ -115,3 +115,406 @@ func TestDirectedCyclesIn(t *testing.T) {
 		}
 	}
 }
+
+var cyclesContainingTests = []struct {
+	g    []intset
+	want [][][]int64
+}{
+	{
+		g: []intset{
+			0: linksTo(1),
+			1: linksTo(2, 7),
+			2: linksTo(3, 6),
+			3: linksTo(4),
+			4: linksTo(2, 5),
+			6: linksTo(3, 5),
+			7: linksTo(0, 6),
+		},
+		want: [][][]int64{
+			0: {{0, 1, 7, 0}},
+			1: {{1, 7, 0, 1}},
+			2: {{2, 3, 4, 2},
+				{2, 6, 3, 4, 2}},
```

Comment:
I'd prefer these kinds of cases to be on a single line; they're not so long that they need two.